File: leetcode/golang/350.两个数组的交集II.go
func intersect(nums1 []int, nums2 []int) []int {
    d := make(map[int]int);
    ret := make([]int, 0);

    for _, item := range nums1 {
        if _, ok := d[item];!ok {
            d[item] = 0;
        }
        d[item] += 1;
    }

    for _, num := range nums2 {
        if _, ok := d[num];ok {
            if d[num] > 0 {
                ret = append(ret, num);
                d[num] -= 1;
            }
        }
    }
    return ret;
}

File: leetcode/python/350.两个数组的交集II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        ret = []
        d = dict()
        for num in nums1:
            if num not in d:
                d[num] = 0
            d[num] += 1
        
        for num in nums2:
            if num in d and d[num] > 0:
                ret.append(num)
                d[num] -= 1
        
        return ret